docs/docs/how-to/installing-shadcn.md

```diff
@@ -0,0 +1,347 @@
+    --primary-foreground: 222.2 47.4% 1.2%;
+ 
+    --secondary: 222.2 47.4% 11.2%;
+    --secondary-foreground: 210 40% 98%;
+ 
+    --destructive: 0 63% 31%;
+    --destructive-foreground: 210 40% 98%;
+ 
+    --ring: 216 34% 17%;
+ 
+    --radius: 0.5rem;
+  }
+}
+ 
+@layer base {
+  * {
+    @apply border-border;
+  }
+  body {
+    @apply bg-background text-foreground;
+    font-feature-settings: "rlig" 1, "calt" 1;
+  }
+}
+```
+
+### 7. Add a cn helper
+
+Shad uses a `cn` helper to make it easier to conditionally add Tailwind CSS classes.
+
+Within the `web/src` directory, create a folder called `lib`, with a file inside called `utils.ts`
+
+Add the following code:
+```ts
+// web/src/lib/utils.ts
+import { clsx, type ClassValue } from 'clsx'
+import { twMerge } from 'tailwind-merge'
+
+export function cn(...inputs: ClassValue[]) {
+  return twMerge(clsx(inputs))
+}
+```
+
+### 8. That's it! ... Kind of. If you want to use ShadCN's command line interface (CLI) keep reading.
+
+If you look at [ShadCN's documentation for their CLI tool](https://ui.shadcn.com/docs/cli), you're supposed to run 
+
+```sh
+npx shadcn-ui@latest init
+```
+
+The problem is that this command installs the dependencies for ShadCN, adds the `cn` util, configures `tailwind.config.js`, and CSS variables for the project, and we already did all those steps! In fact, if you try running that command anyway, it will tell you that Tailwind does not exist and you need to install that first.
+
+One thing that the `init` command does, that we do need to add is a `components.json` file. This file tells ShadCN CLI where everything is located within our project.
+
+Within our `web` directory, create a `components.json` file. The contents for this file is [documented here](https://ui.shadcn.com/docs/components-json). Inside add:
+
+```json
+{
+  "$schema": "https://ui.shadcn.com/schema.json",
+  "aliases": {
+    "components": "src/components",
+    "ui": "src/components/ui",
+    "utils": "src/lib/utils"
+  },
+  "style": "default",
+  "tailwind": {
+    "baseColor": "neutral",
+    "config": "./config/tailwind.config.js",
+    "css": "./src/index.css",
+    "cssVariables": true,
+    "prefix": "tw-",
+    "rsc": false,
+    "tsx": true
+  }
+}
+```
+
+:::note
+A few things worth noting:
+- You might have a different `style` value, but I used the `default` styles when I went through the manual process.
+- The `aliases` section defines all the path names you'll use. I added a `ui` folder to the `web/src/components` directory. To keep the ShadCN components separate from my custom components.
+- ShadCN won't create a Storybook and test file, like the Redwood generators will.
+- Set `tsx` to `true` if you're using TypeScript, `false` if you're using JavaScript.
+:::
+
+### 9. Add a script to your `package.json`
+
+Within the `package.json` file, in the root of your project, add a `scripts` section:
+
+```json
+"scripts": {
+  "shad": "cd web && npx shadcn-ui@latest add"
```

Comment:
The CLI has been updated recently https://ui.shadcn.com/docs/cli, I think this command might need to change to below?

`"shad": "cd web && npx shadcn@latest add"`

Reply:
They also provide `--cwd` option, in my project I use:
```json
"shad": "npx shadcn@latest add -c web"
```

Reply:
https://github.com/redwoodjs/rw-shad sets `--cwd` for you 🙂

Reply:
@Tobbe  Thanks for sharing, haven't seen this yet. I'll learn more about it, probably that's what I was looking for!

But I just went through the process of setting up shadcn for a Redwood project following manual instructions, it wasn't straightforward. And frankly I'm not really satisfied with the final configuration either. But to clean up rough edges, it looks like shadcn cli itself should add support for Redwood. Please, let me know if you look for some help here, while memories are still fresh.

Reply:
@antonmoiseev with https://github.com/redwoodjs/setup-rw-shad and https://github.com/redwoodjs/rw-shad I have tried to give you a setup that is more aligned with the conventions and guidelines we have for Redwood projects. For example I rename all components to start with a capital letter, like we do in RW. And I lint/format all newly added components to conform to RW style rules, etc. 

If you have the time, I'd really appreciate if you try it out and let me know what you think